private/bufpkg/bufmodule/bufmoduleapi/uploader.go

```diff
@@ -111,51 +111,109 @@ func (a *uploader) Upload(
 		return nil, err
 	}
 
+	var modules []*modulev1.Module
 	if uploadOptions.CreateIfNotExist() {
-		if err := a.createContentModulesIfNotExist(
+		// We must attempt to create each module one at a time, since CreateModules is atomic.
```

Comment:
The fact that `CreateModules` is atomic isn't really the reason this doesn't work - in fact, we'd actually prefer to create all the non-existent modules in an atomic manner. The issue is that if any given module already exists, then we get a not-exists error for all modules passed.